catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/stage/BroadcastForClonesRegressionTest.java

```diff
@@ -84,8 +85,10 @@ public void testIfBroadcastsAreReceivedByClones() {
 	@Test
 	public void testIfClonesBroadcastReceiversAreRemovedOnRestart() {
 		baseActivityTestRule.launchActivity(null);
+		IdlingRegistry.getInstance().register(baseActivityTestRule.getActivity().idlingResource);
```

Comment:
That's not quite correct: The android.support.test library is no longer maintained. If we just use the androidx version instead (androidx.test.espresso.IdlingRegistry) we should be good.

We use Idling Resources for test synchronization and to avoid Thread.sleep:
https://developer.android.com/training/testing/espresso/idling-resource

Reply:
basically we are telling espresso that it should wait until the app is idle before executing the next step of the test. An AndroidX PR is approved and just waiting for a merge

Reply:
Nice, we can rebase after that PR was merged and change the import to androidx 👍